### ISSUE TITLE
Regex-parsing a tuple struct with named capture groups by field identity fails to compile

### ISSUE THREAD

**Issue by coriolinus** [frozenlib/parse-display#15]
Example:

```rust
#[derive(Debug, parse_display::Display, parse_display::FromStr)]
#[display("{0:>3} {1:>3} {2:>3}")]
#[from_str(regex = r"(?P<0>\d+)\s+(?P<1>\d+)\s+(?P<2>\d+)")]
struct Triangle(u64, u64, u64);

#[test]
fn parses_compact() {
    "1 2 3".parse::<Triangle>().unwrap();
}

#[test]
fn parses_broad() {
    "  1   2   3".parse::<Triangle>().unwrap();
}
```

This produces the following error:

```
error: regex parse error:
    (?P<_>\d+)\s+(?P<_>\d+)\s+(?P<_>\d+)
        ^            ^
error: duplicate capture group name
 --> src/lib.rs:3:20
  |
3 | #[from_str(regex = r"(?P<0>\d+)\s+(?P<1>\d+)\s+(?P<2>\d+)")]
  |
```

The problem appears to stem from this line of code:

https://github.com/frozenlib/parse-display/blob/3cc7c66e6dad87cc2c090af6924d90b43e5dfe15/parse-display-derive/src/lib.rs#L340

That replaces all digits within the regex with underscores. This breaks field identification for tuple structs.

**Comment by coriolinus** [comment]
Current workaround is just to use a standard struct instead of a tuple struct:

```rust
struct Triangle {
  a: u64,
  b: u64,
  c: u64,
}
```

This works fine, but it's really suboptimal.

**Comment by frozenlib** [comment]
Thanks for the detailed report.

I have now fixed this issue in 223c2c7957eb45f369ca33e51bd449dae1e7d403.

**Comment by coriolinus** [comment]
Thanks! Any idea when that commit might appear in a new release?

**Comment by frozenlib** [comment]
I have now published version 0.5.0 which includes this fix.

**Comment by coriolinus** [comment]
Thanks!
